Welcome to the website for *Applied Mathematics and Informatics In Drug
Discovery* (AMIDD), the course series running at the Department of Mathematics
and Informatics, University of Basel in the fall semester 2021.

The course series introduces interdisciplinary research in drug discovery with
mathematics as the language and informatics as the tool. We have a diverse and
lively class room that learn together and from each other: every year about two
third students of the class study mathematics or computer science, while other
students study physics, chemistry, (computational) biology, pharmacy, and other
fields such as epidemiology and medicine.

More information on the course can be found at [the course directory of the University
Basel](https://vorlesungsverzeichnis.unibas.ch/de/recherche?id=258849).

## Table of content

- [Lectures with Zoom](#lectures-with-zoom)
- [Time and place](#time-and-place)
- [Course material and licensing](#course-material-and-licensing)
- [Pre-course survey](#pre-course-survey)
- [Syllabus](#syllabus)
  * [1. Drug discovery: an overview](#1-drug-discovery-an-overview) (24.09.2021)
  * [2. The central dogma and drug discovery](#2-the-central-dogma-and-drug-discovery) (01.10.2021)
  * [3. Biological sequence analysis](#3-biological-sequence-analysis) (08.10.2021)
  * [4. From sequences to structures](#4-from-sequences-to-structures) (15.10.2021)
  * [5. Proteins and ligands](#5-proteins-and-ligands) (22.10.2021)
  * [6. Structure- and ligand-based drug design](#6-structure--and-ligand-based-drug-design) (29.10.2021)
  * [7. From individial interactions to networks](#7-from-individual-interactions-to-networks) (05.11.2021)
  * [8. Omics and cellular modelling](#8-omics-and-cellular-modelling) (12.11.2021)
  * [9. PKPD modelling](#9-pkpd-modelling) (19.11.2021)
  * [10.*Dies academicus - optional Ask Me Anything session* (26.11.2021)](#10-dies-academicus---optional-ask-me-anything-session)
  * [11. Guest-speaker session](#11-guest-speaker-session) (03.12.2021)
- [Student presentation topics and reference papers](#student-presentation-topics-and-reference-papers)
  * [12. Student presentation (I)](#12-student-presentation-i) (10.12.2021)
      * [Machine learning and artificial intelligence in drug discovery](#MachineLearningAI)
      * [Causal inference in disease understanding and drug discovery](#CausalInference)
  * [13. Student presentation (II)](#13-student-presentation-ii) (17.12.2021)
      * [Drug discovery for SARS-Cov-2](#SARS-COV-2) presented by the Almeida team
      * [Mathematical and computational modelling](#Modelling) presented by the Euler team.
- [End-term project](#end-term-project)
- [Assessment](#assessment)
- [Further questions or suggestions?](#further-questions-or-suggestions)
- [Archives of past courses](#archives-of-past-courses)

## Lectures with Zoom

Due to the coronavirus pandemic, the course in 2021 will take place online with
*Zoom* exclusively. The link to join the Zoom meeting is:
[https://unibas.zoom.us/j/68803401669](https://unibas.zoom.us/j/68803401669).
The passcode for the Zoom meeting is shared among registered students via
emails.

## Time and place

The lecture takes place on Fridays between 12:15 and 14:00 on Zoom. The meeting
will be active between 12:00 and 14:00, so that any questions can be addressed
before the lecture.

## Course material and licensing

Course material, including lecture notes, slides, and reading material, are
shared on the course's web site, [AMIDD.ch](http://amidd.ch), unless otherwise
specified in the course.

All Zoom sessions are recorded and distributed among attendees.

All course material, unless otherwise stated, is shared under the Creative
Commons (CC-BY-SA 4.0) license.

## Pre-course survey

Prior to attending the first session, please [fill out the voluntary pre-course
survey](https://forms.gle/Qd6PdWv5LuwZDw6w7). It helps me to shape the course to
meet your needs.

## Syllabus

### 1. Drug discovery: an overview

* [Slides]({{site.assetbaseurl}}{% link assets/2021/01/AMIDD-2021-01-Intro.pdf %})
* [Anonymous Post-lecture Survey #1](https://forms.gle/nahRRtbDPzZTKKAz5)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/YrrIKSRC_07QxZjasj42bUsa21w48VlWEKDzZJrWU9rfPTs0IF2zHazwMphBAf3W.QIp8zvuLjEGI4-s6)
* Material for offline activities (see slides 34-36)
    * [Link to the video](https://www.ibiology.org/human-disease/herceptin/) on
      the discovery and development of Herceptin, by Susan Desmond-Hellmann
    * [Principles of early drug discovery]({{site.assetbaseurl}}{% link
      assets/2019/01/Principles-DD-Hughes.pdf %}) by Hughes *et al.*
    * Submit your answers to the offline activities [here via Google Form](https://forms.gle/R9k7aGVLfd8J4bLr6)

### 2. The central dogma and drug discovery

* [Slides]({{site.assetbaseurl}}{% link assets/2021/02/AMIDD-2021-02-TheCentralDogma.pdf %})
* [Anonymous Post-lecture Survey #2](https://forms.gle/YefzD76iwd9LvW3YA)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/Bug4v5SAKSs6-iIgREmBBOKZbX1YSzP1Me6PUVJwrZ4VdNu9xvgQ8iZEjoBuNCvh.roQgCCOtkEROuYAe)
* Material for offline activities (see slides 34-36)
    * Required reading: [Clinical efficacy of a RAF inhibitor needs broad target blockade in BRAF-mutant melanoma]({{site.assetbaseurl}}{% link assets/2021/02/Bollag-Nature-2010.pdf %}) by Bollag *et al.*, Nature 2010.
    * Optional reading: [A comprehensive map of molecular drug targets]({{site.assetbaseurl}}{% link assets/2021/02/Santos-NRDD-2017.pdf %}) by Santos *et al.*, Nature Reviews Drug Discovery, 2017
* Submit your answers for Offline Activities [here via Google Form](https://forms.gle/B1gp5gnWW6koev59A)

### 3. Biological sequence analysis

* [Slides]({{site.assetbaseurl}}{% link assets/2021/03/AMIDD-2021-03-BiologicalSequenceAnalysis.pdf %})
* [Anonymous Post-lecture Survey #3](https://forms.gle/EAfmEEG7JtCnW1m7A)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/8DWc9YApVBl0nFEt-rN_uZXMcERSpjl7gq267pNbqhekCK36qlxLLj8PfXr1D4EZ.WdSsS9Hp5glCshyW)
* Material for offline activities
  * [Handout for lecture 2 and lecture 3]({{site.assetbaseurl}}{% link assets/2021/02/AMIDD-Lecture2-Handout.pdf %}), which contains genetic codes, information on amino acids, and offline exercises (which are due in two weeks, before Lecture 5).
  * Required reading: [Discovery of a selective inhibitor of oncogenic B-Raf kinase with potent antimelanoma activity]({{site.assetbaseurl}}{% link assets/2021/03/Tsai-BRAF-PNAS-2008.pdf %}) by Tsai *et al.*, PNAS 2008.
  * Optional reading: [The Tangled History of MRNA Vaccines](https://doi.org/10.1038/d41586-021-02483-w) by Elie Dolgin, Nature 2021.
* Submit your answers for offline activities [here via Google Form](https://forms.gle/VzGGbXBvy9ZhTSVK8). *They are due only in two weeks, before Lecture 5*, because we will cover the exercises of biological sequence analysis only in Lecture 4.


### 4. From sequences to structures

* [Slides]({{site.assetbaseurl}}{% link assets/2021/04/AMIDD-2021-04-From-Sequences-to-Structures.pdf %})
* [Anonymous Post-lecture Survey #4](https://forms.gle/ajehjtKYPngji4yX7)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/TsMJr5JRaEzkWWbAILLuLjbOglmIp_ZgqHEfC4UB7RALJJ3mloGq309mmlqMwGo.j3qOVERQBN4iXSYo)
* Material for offline activities: the same as lecture 3

### 5. Proteins and ligands

* [Slides]({{site.assetbaseurl}}{% link assets/2021/05/AMIDD-2021-05-proteins-and-ligands.pdf %})
* [Anonymous Post-lecture Survey #5](https://forms.gle/YQRALAV6HiZ9QswZ6)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/j55JpNr3ZA0DqKpljQJIAPNwZ9TsCqNVzYYAxl0jsVGWetzVfg566bJvNkf6ORTU.wy2fJD9Q8E9n67zn)
* Material for offline activities:
   * Required reading: [*Evaluation of the Biological Activity of Compounds: Techniques and Mechanism of Action Studies*]({{site.assetbaseurl}}{% link assets/2021/05/BiologicalActivity-Dougall-Unitt.pdf%}) by Iain G. Dougall and John Unitt, chapter two of the book *The Practice of Medicinal Chemistry*. **To answer offline-activity questions**, it is required to read pages 15-22 (1-8 of the 29 pages in total, before section &lsquo;*4. Types of Enzyme Inhibition and Their Analysis*&rsquo;), page 27 (section 6A), and pages 34-37 (*Assay Biostatistics*). The rest is optional reading.
    * Optional reading:
        * [Mathematics techniques in structural biology]({{site.assetbaseurl}}{% link assets/2021/05/JRQuine-MathBiophysicsBook.pdf %}) by John R. Quine.  Recommended booklet for students interested in applications of mathematics in determining structures of DNA and proteins without or with drugs.
        * Mullard, Asher. “What Does AlphaFold Mean for Drug Discovery?” Nature Reviews Drug Discovery 20, no. 10 (September 14, 2021): 725–27. [Web link](https://doi.org/10.1038/d41573-021-00161-0). [PDF version available here]({{site.assetbaseurl}}{% link assets/2021/05/Mullard-2021-NRDD-AlphaFold-DD.pdf %}).
   * Submit your results here [via Google Form](https://forms.gle/ktcoMPP8ChF3x1eo8)

### 6. Structure- and ligand-based drug design

* [Slides]({{site.assetbaseurl}}{% link assets/2021/06/AMIDD-2021-06-Structure-and-ligand-based-drug-discovery.pdf %})
* [Anonymous Post-lecture Survey #6](https://forms.gle/nBcTT6EdrR1W4JN36)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/873ODLc7hA0jH5okp5MsUF1u2U-JUtendltOGJAE3tgJEBAc-QLLp-q1g9RMGSBV.tiwcTssuRGfEIPei)
* Material for offline activities:
    * You can vote for your presentation topics via [Google Form](https://forms.gle/pov9pfaitWAwVoFS9) by November the first.
    * **Offline activity: required reading**: [*Computational methods in drug discovery*]({{site.assetbaseurl}}{% link assets/2021/06/Sliwoski-PharmacologicalReviews-2014-Computational-Methods-In-Drug-Discovery.pdf %}) by Gregory Sliwoski *et al.*. Please submit your replies to questions [via this Google Form](https://forms.gle/XjbLVi1QG5R1QiyDA).
    * Optional reading: [*An introduction to machine learning*]({{site.assetbaseurl}}{% link assets/2021/06/Badillo-ML-2020.pdf %}) by Badillo *et al.*.


### 7. From individual interactions to networks


* [Slides]({{site.assetbaseurl}}{% link assets/2021/07/AMIDD-2021-07-from-individual-interactions-to-networks.pdf %})
* [Anonymous Post-lecture Survey #7](https://forms.gle/ECBaiYvAdroimxuy7)
* [Recording (passcode shared by emails)](https://unibas.zoom.us/rec/share/PvcZ_6mAP4NL-HXPWwEuaIbVEKPwLkfpB2GXPW1K-VpswWXqhcAh8M0jnkdzpg74.KWScjcWrQOOIf8AF)
* No new offline activity this week, because the activity last week was time consuming. If you are not yet familiar with machine learning, please read the optional reading assigned last week, [*An introduction to machine learning*]({{site.assetbaseurl}}{% link assets/2021/06/Badillo-ML-2020.pdf %}) by Badillo *et al.*

### 8. Omics and cellular modelling

* [Slides]({{site.assetbaseurl}}{% link assets/2021/08/AMIDD-2021-08-omics.pdf %})
* [Anonymous post-lecture survey #8](https://forms.gle/wPWzm7gFCCnAHAaG8)
* [Recording (password shared by emails)](https://unibas.zoom.us/rec/share/chedtOVPqq6Da-bsUlJxo6QSjAWc5sbGYu744hlE-_N1PiVkWNZLi6drV6qZ-rNg.Su8Z8PuibILVaNJR)
* Offline activities
    * Required readings (1): [Biological Image Analysis Primer]({{site.assetbaseurl}}{% link assets/2021/08/Meijering-vanCappellen-2006-BioImage.pdf %}) by Erik Meijering and Gert van Cappellen (2006).
    * Required readings (2): Rudin, Markus, and Ralph Weissleder. 2003. “[Molecular Imaging in Drug Discovery and Development](https://doi.org/10.1038/nrd1007).” Nature Reviews Drug Discovery 2 (2): 123–31. ([PDF version]({{site.assetbaseurl}}{% link assets/2021/08/Rudin-Weissleder-2003-Molecular-imaging.pdf %}))
    * Submit your offline activities with regard to the required reading [here via the Google Form](https://forms.gle/BP8r46B2iWaeMaXL7). Deadline: Friday, November 26th, 2021.
    * Optional reading: [Spatial omics and multiplexed imaging to explore cancer biology](https://www.nature.com/articles/s41592-021-01203-6), a review by Lewis *et al.*

### 9. PKPD modelling

* [Slides]({{site.assetbaseurl}}{% link assets/2021/09/AMIDD-2021-09-PK-PD-modelling.pdf %})
* [Anonymous post-lecture survey #9](https://forms.gle/yiv7Xo3gJH1XDsZu5)
* [Recording (password shared by
  emails)](https://unibas.zoom.us/rec/share/ODJwhBXofea7bsSZ_ErpNqectSGUUburVh-5CW9uZOF0piQLFIOfypm2cVQKGTg.fICSH5sVf_ZdkM3-)
* Offline activities:
    * Required reading: backup slides on non-linear mixed-effect (NLME) models and clinical trials.
    * Optional reading:
        * Davies, Michael, et al.. 2020. “[Improving the Accuracy of Predicted Human Pharmacokinetics: Lessons Learned from the AstraZeneca Drug Pipeline Over Two Decades](https://doi.org/10.1016/j.tips.2020.03.004).” Trends in Pharmacological Sciences 41 (6): 390–408. *A good introduction to prediction of PK profiles in industry*.
        * Jones, H. M., and K. Rowland‐Yeo. 2013. “[Basic Concepts in Physiologically Based Pharmacokinetic Modeling in Drug Discovery and Development](https://doi.org/10.1038/psp.2013.41).” CPT: Pharmacometrics & Systems Pharmacology 2 (8): 63. *A good introduction to PBPK modelling*.


### 10. *Dies academicus - optional Ask Me Anything session*


You can ask me anything in this session, which will be exceptionally *not* recorded.

Besides scientific topics in drug discovery, my experience is that many students are interested in career topics: should I do a PhD or not? Should I consider working in industry? You may this article below interesting if you consider doing a PhD and perhaps doing a postdoc in pharma: Zhang, Jitao David. “[Ten Simple Rules for Doing a Postdoc in Pharma.](https://doi.org/10.1371/journal.pcbi.1008989)” PLOS Computational Biology 17, no. 6 (June 3, 2021): e1008989.

### 11. Guest-speaker session

This year we will welcome our guest speakers Dr. Elif Ozkirimli, Dr. Thomas
Sander, and Dr. Juliane Siebourg-Polster to share with us their industrial
research experience in applied mathematics and informatics in drug discovery.

*Please note that the course starts exceptionally at 12:05*.

* **Dr. Thomas Sander** studied organic chemistry in Marburg, Germany (PhD with
  Prof. R.W. Hoffmann). Then he did a post-doc in cheminformatics with J.B.
  Hendrickson, Brandeis University, Waltham, Massachusetts. After that he worked
  for 5 years at Roche developing applications/databases for drug discovery.
  Dr. Sander joined Actelion as one of the first employees building up and
  shaping the drug discovery software landscape with a growing team for 20
  years. Now he is at Idorsia and mainly works on open-source cheminformatics
  projects.
* **Dr. Juliane Siebourg-Polster** is a biomathematician by training with a PhD in
  computational biology from ETH Zürich. Currently, she works as a
  Biostatistician in pre-clinical research at Roche pRED in Basel. She enjoys
  working at the interface of biology, data analysis and mathematical modeling.
  In pre-clinical drug discovery and development her focus is in statistical
  analysis and modeling of high throughput omics biomarker studies as well as
  experiment design. In addition she engages in teaching statistical methods and
  concepts to colleagues.
* **Dr. Elif Ozkirimli** is the head of data science and advanced analytics at
  Roche. In her current role, she is focusing on the application of machine
  learning (ML), natural-language processing (NLP), advanced mathematical
  modeling and statistical approaches to understand the healthcare professional
  or patient journeys to bring higher patient and healthcare impact. Previously,
  Elif was a professor of chemical engineering at Bogazici University, Turkey,
  and focused on the application of NLP and deep-learning (DL) approaches for
  protein - compound interaction prediction. Elif has a PhD in structural
  biology from Purdue University.

* [Anonymous post-lecture survey, with questions for the guest speakers](https://forms.gle/Duu1MdcnjfEMG6rQ7)
* [Recording (password shared by emails)](https://unibas.zoom.us/rec/share/URGNwPojCpdKvObCgbQ2fGrSimf4g7I9SWysxby9_KMiLPYh_KbsJ0USxLizJHDN.foFdzsu0b4VUxHLx)
* [*Translator in Science*, the presentation by Dr. Juliane Sieborg-Polster]({{site.assetbaseurl}}{% link assets/2021/11/20211203-AMIDD-JulianeSiebourgPolster-Translator-in-science.pdf %})
* [*Needles in a haystack: From coordinates to strings, from structural biology to NLP and AI*, the presentation by Dr. Elif Ozkirimli]({{site.assetbaseurl}}{% link assets/2021/11/20211203-AMIDD-ElifOzkirimli-Needles-in-a-Haystack.pdf %})

## Student presentation topics and reference papers

You can vote for your presentation topics via [Google Form](https://forms.gle/pov9pfaitWAwVoFS9) by November the first.

In HS 2021, our class will present on four topics: machine learning, causal
inference, drug discovery for SARS-Cov-2, and mathematical and computational
modelling. Each team is expected to deliver a talk of 30 minutes, which is
followed by 10 minutes of Q&A.

[Please find a table here]({{site.assetbaseurl}}{% link
assets/2021/AMIDD-2021-presentation-list.pdf %}) listing teams, the presentation
topics, and the time slots in the PDF format (updated on 14.11.2021).

Unfortunately we cannot satisfy the availability of everyone. I hope you can
understand the difficulty, and arrange within each team so that all members can
contribute to the presentation, whether she or he is available during the
presentation is not.

Below we list the topics and relevant references. Each group is expected to
select 2-4 references from the given ones and discuss them. Additional
references are welcome.

### 12. Student presentation (I)

<a name="MachineLearningAI"></a>

* [Zoom recording](https://unibas.zoom.us/rec/share/cu5undOxlvKYvCa1JBSf6BE4weX7iWDy38THK7YHNuINzBiQgwHYLjEsGpsGtBU.s6OdY7QXX1MTsm48)

#### *Machine learning and artificial intelligence in drug discovery*, presented by the Turing team

Reference papers:

1. Bender, Andreas, and Isidro Cortés-Ciriano. “Artificial Intelligence in Drug Discovery: What Is Realistic, What Are Illusions? Part 1: Ways to Make an Impact, and Why We Are Not There Yet.” Drug Discovery Today, December 17, 2020. [https://doi.org/10.1016/j.drudis.2020.12.009](https://doi.org/10.1016/j.drudis.2020.12.009).
2. Bender, Andreas, and Isidro Cortes-Ciriano. “Artificial Intelligence in Drug Discovery: What Is Realistic, What Are Illusions? Part 2: A Discussion of Chemical and Biological Data.” Drug Discovery Today 26, no. 4 (April 1, 2021): 1040–52. [https://doi.org/10.1016/j.drudis.2020.11.037](https://doi.org/10.1016/j.drudis.2020.11.037).
3. Barredo Arrieta, Alejandro, Natalia Díaz-Rodríguez, Javier Del Ser, Adrien Bennetot, Siham Tabik, Alberto Barbado, Salvador Garcia, et al. “Explainable Artificial Intelligence (XAI): Concepts, Taxonomies, Opportunities and Challenges toward Responsible AI.” Information Fusion 58 (June 1, 2020): 82–115. [https://doi.org/10.1016/j.inffus.2019.12.012](https://doi.org/10.1016/j.inffus.2019.12.012).
4. Stokes, Jonathan M., Kevin Yang, Kyle Swanson, Wengong Jin, Andres Cubillos-Ruiz, Nina M. Donghia, Craig R. MacNair, et al. “A Deep Learning Approach to Antibiotic Discovery.” Cell 180, no. 4 (February 20, 2020): 688-702.e13. [https://doi.org/10.1016/j.cell.2020.01.021](https://doi.org/10.1016/j.cell.2020.01.021).
5. Basile, Anna O., Alexandre Yahi, and Nicholas P. Tatonetti. “Artificial Intelligence for Drug Toxicity and Safety.” Trends in Pharmacological Sciences 40, no. 9 (September 1, 2019): 624–35. [https://doi.org/10.1016/j.tips.2019.07.005](https://doi.org/10.1016/j.tips.2019.07.005).
6. Ietswaart, Robert, Seda Arat, Amanda X. Chen, Saman Farahmand, Bumjun Kim, William DuMouchel, Duncan Armstrong, Alexander Fekete, Jeffrey J. Sutherland, and Laszlo Urban. “Machine Learning Guided Association of Adverse Drug Reactions with in Vitro Target-Based Pharmacology.” EBioMedicine, June 18, 2020, 102837. [https://doi.org/10.1016/j.ebiom.2020.102837](https://doi.org/10.1016/j.ebiom.2020.102837).


<a name="CausalInference"></a>

#### *Causal inference in disease understanding and drug discovery*, presented by the Hill team

Reference papers and software:

1. Hill, Austin Bradford. “The Environment and Disease: Association or Causation?” Proceedings of the Royal Society of Medicine 58, no. 5 (May 1965): 295–300.
2. Shrier, Ian, and Robert W. Platt. “Reducing Bias through Directed Acyclic Graphs.” BMC Medical Research Methodology 8, no. 1 (October 30, 2008): 70. [https://doi.org/10.1186/1471-2288-8-70](https://doi.org/10.1186/1471-2288-8-70).
3. Evans, David M., and George Davey Smith. “Mendelian Randomization: New Applications in the Coming Age of Hypothesis-Free Causality.” Annual Review of Genomics and Human Genetics 16, no. 1 (2015): 327–50. [https://doi.org/10.1146/annurev-genom-090314-050016](https://doi.org/10.1146/annurev-genom-090314-050016).
4. Prosperi, Mattia, Yi Guo, Matt Sperrin, James S. Koopman, Jae S. Min, Xing He, Shannan Rich, Mo Wang, Iain E. Buchan, and Jiang Bian. “Causal Inference and Counterfactual Prediction in Machine Learning for Actionable Healthcare.” Nature Machine Intelligence 2, no. 7 (July 2020): 369–75. [https://doi.org/10.1038/s42256-020-0197-y](https://doi.org/10.1038/s42256-020-0197-y).
5. Bica, Ioana, Ahmed M. Alaa, Craig Lambert, and Mihaela van der Schaar. “From Real-World Patient Data to Individualized Treatment Effects Using Machine Learning: Current and Future Methods to Address Underlying Challenges.” Clinical Pharmacology & Therapeutics 109, no. 1 (2021): 87–100. [https://doi.org/10.1002/cpt.1907](https://doi.org/10.1002/cpt.1907).
6. [Tutorial of DoWhy](https://microsoft.github.io/dowhy/).

### 13. Student presentation (II)

<a name="SARS-COV-2"></a>

#### *Drug-discovery for SARS-Cov-2*, presented by the Almeida team

1. Tummino, Tia A., Veronica V. Rezelj, Benoit Fischer, Audrey Fischer, Matthew J. O’Meara, Blandine Monel, Thomas Vallet, et al. “Drug-Induced Phospholipidosis Confounds Drug Repurposing for SARS-CoV-2.” Science 373, no. 6554 (July 30, 2021): 541–47. [https://doi.org/10.1126/science.abi4708](https://doi.org/10.1126/science.abi4708).
2. Delft, Frank von, Mark Calmiano, John Chodera, Ed Griffen, Alpha Lee, Nir London, Tatiana Matviuk, Ben Perry, Matt Robinson, and Annette von Delft. “A White-Knuckle Ride of Open COVID Drug Discovery.” Nature 594, no. 7863 (June 2021): 330–32. [https://doi.org/10.1038/d41586-021-01571-1](https://doi.org/10.1038/d41586-021-01571-1).
3. Mei, Miao, and Xu Tan. “Current Strategies of Antiviral Drug Discovery for COVID-19.” Frontiers in Molecular Biosciences 8 (2021): 310. [https://doi.org/10.3389/fmolb.2021.671263](https://doi.org/10.3389/fmolb.2021.671263).
4. Kwok, Andrew J., Alex Mentzer, and Julian C. Knight. “Host Genetics and Infectious Disease: New Tools, Insights and Translational Opportunities.” Nature Reviews Genetics 22, no. 3 (March 2021): 137–53. [https://doi.org/10.1038/s41576-020-00297-6](https://doi.org/10.1038/s41576-020-00297-6).
5. Heinz, Franz X., and Karin Stiasny. “Distinguishing Features of Current COVID-19 Vaccines: Knowns and Unknowns of Antigen Presentation and Modes of Action.” Npj Vaccines 6, no. 1 (August 16, 2021): 1–13. [https://doi.org/10.1038/s41541-021-00369-6](https://doi.org/10.1038/s41541-021-00369-6).
6. Relevant posts in [In The Pipeline](https://www.science.org/blogs/pipeline), for instance [Pfizer's Good News Is the World's Good News](https://www.science.org/content/blog-post/pfizer-s-good-news-world-s-good-news) and [Covid Therapeutic Trial Recruitment](https://www.science.org/content/blog-post/covid-therapeutic-trial-recruitment).

<a name="Modelling"></a>

#### *Mathematical and computational modelling*, presented by the Euler team

1. Handel, Andreas, Nicole L. La Gruta, and Paul G. Thomas. “Simulation Modelling for Immunologists.” Nature Reviews Immunology 20, no. 3 (March 2020): 186–95. [https://doi.org/10.1038/s41577-019-0235-3](https://doi.org/10.1038/s41577-019-0235-3).
2. Jones, H. M., and K. Rowland‐Yeo. “Basic Concepts in Physiologically Based Pharmacokinetic Modeling in Drug Discovery and Development.” CPT: Pharmacometrics & Systems Pharmacology 2, no. 8 (August 1, 2013): 63. [https://doi.org/10.1038/psp.2013.41](https://doi.org/10.1038/psp.2013.41).
3. Gieschke, R., and J. L. Steimer. “Pharmacometrics: Modelling and Simulation Tools to Improve Decision Making in Clinical Drug Development.” European Journal of Drug Metabolism and Pharmacokinetics 25, no. 1 (March 2000): 49–58. [https://doi.org/10.1007/BF03190058](https://doi.org/10.1007/BF03190058).
4. Sancho-Araiz, Aymara, Victor Mangas-Sanjuan, and Iñaki F. Trocóniz. “The Role of Mathematical Models in Immuno-Oncology: Challenges and Future Perspectives.” Pharmaceutics 13, no. 7 (July 2, 2021): 1016. [https://doi.org/10.3390/pharmaceutics13071016](https://doi.org/10.3390/pharmaceutics13071016).
5. Elmokadem, Ahmed, Matthew M. Riggs, and Kyle T. Baron. “Quantitative Systems Pharmacology and Physiologically-Based Pharmacokinetic Modeling With Mrgsolve: A Hands-On Tutorial.” CPT: Pharmacometrics & Systems Pharmacology 8, no. 12 (2019): 883–93. [https://doi.org/10.1002/psp4.12467](https://doi.org/10.1002/psp4.12467).
6. Lu, James, Brendan Bender, Jin Y. Jin, and Yuanfang Guan. “Deep Learning Prediction of Patient Response Time Course from Early Data via Neural-Pharmacokinetic/Pharmacodynamic Modelling.” Nature Machine Intelligence 3, no. 8 (August 2021): 696–704. [https://doi.org/10.1038/s42256-021-00357-4](https://doi.org/10.1038/s42256-021-00357-4).

## End-term project

All participants are expected to finish the end-term project either individually
or in a team of two people. She or he or the team shall choose *one* concept
form the list below, upon which they will write a short essay introducing the
concept to non-experts, with examples and ideally applications in drug
discovery.

The essay should have an abstract (fewer than 200 words), a list of references,
and a main text that does not exceed 3,000 words. Visual elements like table and
figures that help readers understand the concept are welcome but not obligatory.
In case of a team work, the contribution of the two people should be specified.

Essays in literature-programming styles, combining texts and source code with
Jupyter notebook or Rmarkdown, are welcome. For such essays the source code does
not count to the total words.

List of candidate topics:

1. [Simple/Statistics] Statistical power and sample-size calculation
2. [Simple/Machine learning] Principal component analysis (PCA)
3. [Simple/Algorithm] The Viterbi algorithm
4. [Simple/Mathematical modelling] The Lotka-Volterra predator-prey model
5. [Moderate/Statistics] Exploratory factor analysis
6. [Moderate/Machine learning] Classification and regression trees
7. [Moderate/Causal inference] Directed acyclic graphs (DAGs) for causal inference
8. [Challenging/Causal inference] Mendelian randomization
9. [Challenging/Algorithm] Community detection with Leiden clustering
10. [Challenging/Statistics] Simpson's paradox and mixed models

The choice of topic and the team composition is to be submitted via [this Google Form of voting for the project topic](https://forms.gle/VUTeFqsbu3yoF2DDA) by *Friday, December the 3rd*. Two people working together need to vote just once. The deadline for submitting the essay by email is *January 14th, 2022*. No extension is possible.

You can view the topics of choice [here]({{site.assetbaseurl}}{% link assets/2021/AMIDD-2021-EndTerm-Public.pdf %})

Once the essays are submitted, they will be shared with the whole class for open review and joint learning.

## Assessment

The final note is given by participation (40%), presentation (30%), and project
work (30%).

## Further questions or suggestions?


Please contact the lecturer, Jitao David Zhang, at [jitao-david.zhang@unibas.ch](mailto:jitao-david.zhang@unibas.ch).


## Archives of past courses

* [AMIDD 2020]({{site.baseurl}}{% link archive/README-2020.md %})
* [AMIDD 2019]({{site.baseurl}}{% link archive/README-2019.md %})
