Introduction to Applied Mathematics and Informatics in Drug Discovery (iAMIDD)
===

[Jitao David Zhang](mailto:jitao_david.zhang@roche.com), Pharma Research and Early Development, Roche Innovation Center Basel, Pharmaceutical Sciences, Grenzacherstrasse 124, 4070 Basel, Switzerland


Applied Mathematics and Informatics In Drug Discovery (AMIDD) is a course that I am designing. It is expected to be given in the autumn semester 2019 to an audience of university students who are interested in this topic.

The course material is published under the GPL-3 license.

## Motivation

Interdisciplinary research is imperative to drug discovery and disease understanding in the industrial setting. Mathematics and informatics play a central role in orchestrating different fields of research and integrating available knowledge and information to enable decisions. The speed and quality of these decisions have direct impacts on life quality of patients and their families. It is therefore of common interest to systematically review concepts, tools, and approaches of applied mathematics and informatics in drug discovery to encourage innovation. 

## Goals 

It is hoped that with an introduction to mathematics and informatics in drug discovery, and especially the motivation and the impacts of their applications, the students are motivated to deepen their knowledge in relevant fields. Equally important, the students are encouraged to apply what they learn to solve open challenges in drug discovery and to gain new insights.

## Audience

This course is designed for all students who are interested in applied mathematics and informatics and their applications in modern drug discovery. Though no prerequisite courses are obligatory, elementary understanding of statistics and ordinary differential equations are helpful. Knowledge and proficiency in at least one programming language (preferably C/C++, Java, R, Python, or Julia) is very helpful to try real-world problems.

## Content

The course is designed to be part of the curriculum of applied mathematics for undergraduates to give a high-level overview of applications of mathematics and informatics tools in drug discovery. Therefore, while key concepts and principles are introduced, almost all subjects can only be briefly touched. Intensive investigations into some subjects, for instance bioinformatics and computational biology, are planned as follow-up courses.

The course consists of 14 sessions (two hours each) and a final oral examination. The sessions will cover an introduction to drug discovery (1 session), bioinformatics and computational biology (4 sessions), chemoinformatics and computer-aided drug design (3 sessions), statistical and mathematical modelling (4 sessions), and key approaches to translate candidates into drugs such as biomarkers and clinical trials (2 sessions).  The sessions are so arranged that they roughly reflect the general drug discovery pipeline, starting from target assessment, transiting via screening, lead identification and optimization, preclinical safety evaluation, PK/PD modelling, clinical trials to regulatory approval.

## Syllabus

1. Part I: Introduction
    1. Introduction to drug discovery
1. Part II: Bioinformatics and computational biology
    1. Biological sequence analysis
    1. Next-generation sequencing
    1. From RNA to antisense oligonucleotides
    1. Protein: from sequence to structure
1. Part III: Chemoinformatics and computer-aided drug design
    1. Chemical structure representation and search
    1. Proximity and molecular interactions
    1. Molecular descriptors and QSAR
1. Part IV: Statistical and mathematical modelling
    1. Causal analysis and inference
    1. Computational and translational enzymology
    1. Mechanistic cardiac modelling for drug safety evaluation
    1. Pharmacokinetics/Pharmacodynamics (PK/PD) modelling
1. Part V: From candidates to drugs (and back)
    1. Emerging biomarkers: imaging and digital biomarker
    1. Clinical trials and pharmacometrics

## Acknowledgement

I got great input and support from numerous colleagues to design and implement the course. Especially I would like to thank 

* Martin Ebeling, Fabian Birzele, and Iakov Davydov for the bioinformatics part.
* Lisa Sach-Peltason, Christian Kramer and Michael Reutlinger for the chemoinformatics and drug design part.
* Manfred Kansy and Matthias Nettekoven for chemistry.
* Arne Rufer, Ken Wang, Norman Mazer, Neil Parrot Jones, Francois Mercier, and Gonzalo Duran-Pacheco for the modelling and pharmacology part.
* Markus von Kienlin, Andreas Bruns and Kasper Rufibach for the biomarker and clinical part


