---
title: "Omics and next-generation sequencing"
author: "Jitao david Zhang"
date: "`r format(Sys.Date(), '%b %d, %Y')`"
fontsize: 10pt
output: 
  beamer_presentation:
    includes:
      in_header: mypreamble.tex
    toc: true
    colortheme: dolphin
    fig_height: 4
    fig_width: 4
    fonttheme: structurebold
    theme: CambridgeUS
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# Omics

## Omics in the view of the central dogma

## Genomics

* Three billion bases in human genome in the form of \texttt{ATGCGACT...}.
* Genomic information, together with environmental factors, determine *traits*, including risk to development disease and likelihood to suffer from diseases
    * APOE: associated with risk of Alzheimer
    * HER2: associated with response to HER2-blocking anti-cancer therapy
    * (TODO: another example of CYP)

\takehome{Genomics among others helps to stratify patients}

## GWAS studies

* GWAS = Genome-while association studies
* Explore genotype-phenotype associations in a large population
* One way to identify target genes, conditional on functional validation

## Transcriptomics

* About 20,000 genes in the human genome, both protein-coding and non-coding.
* Genes in the genome are transcribed as message RNAs (mRNAs), in the form of \texttt{AUGGACT}
* Much information are available by profiling the transcriptome by sequencing
     * Nucleotide sequence
     * Abundance

\takehome{Transcriptomics among others characterizes drug candidates}

## Proteomics

* About 20,000 proteins in the human genome, which are translated from mRNAs into proteins.
* Proteomics can quantify expression levels of proteins, but also post-translational modifications (PTMs), such as phosphorylation, ubiquitination, etc.
\takehome{Proteomics among others identifies biomarkers}

# Next-generation sequencing

## Principles

\takehome{NGS generate reads from DNA}

## Mapping reads to the genome with BWA

\takehome{BWA is an efficient way to map reads to the genome}

## Mapping reads to the transcriptome with graph models

\takehome{Graph models allow fast mapping of reads to transcriptome}

## Model gene expression data with negative-binomial distributions and generalized linear models

\takehome{NB+GLM model NGS gene expression and differential expression}

## Negative-binomial distributions

Two understanding of negative-binomial distributions

## Linear models

## Generalized linear models

## Gene-set enrichment analysis

\takehome{Competative and self-containing methods are available for GSEA}

# Conclusions

## Omics complement classical pharmacological studies of drug candidates

## Other topics

* Variants of NGS to study chromatin structure, epigenetics, \textit{etc.}
* Mass spectrometry, flow cytometry and Cytometry of the Masses (CyTOF)
* Metabolomics
* Single-cell genomics, transcriptomics and proteomics
* Spatial transcriptomics
