---
title: "Biological sequence analysis in drug discovery"
author: "Jitao David Zhang"
date: "`r format(Sys.Date(), '%b %d, %Y')`"
fontsize: 10pt
output: 
  beamer_presentation:
    includes:
      in_header: mypreamble.tex
    toc: true
    fig_height: 4
    fig_width: 4
    incremental: true
    theme: metropolis
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(Rgraphviz)
library(RColorBrewer)
library(ribiosPlot)
```

# The Central Dogma of Molecular Biology

## The Central Dogma: A law of biological information flow

\begincols
  \begincol{.5\textwidth}
    Central dogma
  \endcol
  \begincol{.5\textwidth}
```{r cdgraph, fig.height=4, fig.width=4}
cdg <- new("graphNEL", nodes=c("DNA", "RNA", "Protein"), edgemode="directed")
cdg <- addEdge("DNA", "RNA", cdg, 1)
cdg <- addEdge("DNA", "DNA", cdg, 1)
cdg <- addEdge("RNA", "RNA", cdg, 1)
cdg <- addEdge("RNA", "DNA", cdg, 1)
cdg <- addEdge("RNA", "Protein", cdg, 1)

geneticColors <- brewer.pal(9, "Pastel1")
nAttrs <- list()
nAttrs$fillcolor <- list("DNA"=geneticColors[1],
               "RNA"=geneticColors[2],
               "Protein"=geneticColors[3])
eAttrs <- list()
eAttrs$label <- c("DNA~RNA"="Transcription",
                  "RNA~DNA"="Reverse Transcription",
                  "RNA~Protein"="Translation",
                  "DNA~DNA"="Replication")
eAttrs$style <- c("RNA~DNA"="dotted")

{
  ## compactPar()
  plot(cdg, edgeAttrs=eAttrs, nodeAttrs=nAttrs,
       attrs=list(node=list(fontsize=12, height=1, width=1),
                  edge=list(color="black", arrowsize=0.3),
                  graph=list()),
       recipEdges="distinct")
  ## TODO: (1) dotted in reverse transcription (2) offsets of labels. Are both possible with Rgraphviz?? I am not sure.
}
```
  \endcol
\endcols  
   
\takehome{The CD graph view offers insights for drug discovery}

## Drugs target either nodes or edges of the graph

\takehome{Drugs target either nodes or edges of the CD graph}

## Examples of drugs that target DNA or RNA

\takehome{More and more drugs target DNA/RNA of the CD graph}

## Antisense oligonucleotides

\takehome{Sequence analysis allows designing potent and specific ASOs}

## Sequence analysis is critical for phylogenetic analysis

\takehome{Sequence analysis helps us understanding evolution of sequences}

# DNA/RNA sequence analysis

## DNA: the chemistry and the symbolic representation

\takehome{DNA is a string of characters and often form double helixes}

## RNA: the chemistry and the symbolic representation

\takehome{RNA is a string of characters and is often single-stranded}

## Levenshtein distance

<!-- This method was invented in 1965 by the Russian Mathematician Vladimir Levenshtein (1935-2017).  --> 

---
## This another way to make comments within Rmd file
---

- The minimum number of operations required to transform string $a$ to string $b$ with following operations:
    * Insersion, for instance sit $\textrm{bat}~\rightarrow~\textrm{bait}$ 
    * Deletion, \textit{e.g.} $\textrm{boat}~\rightarrow~\textrm{bot}$
    * Substitution, \textit{e.g.} $\textrm{pig}~\rightarrow~\textrm{big}$
- The Levenshtein distance between two strings $a, b$ of length $|a|$ and $|b|$ respectively is given by $\operatorname{lev}_{a,b}(|a|,|b|)$ where
  $\qquad\operatorname{lev}_{a,b}(i,j) = \begin{cases}
    \max(i,j) & \text{ if } \min(i,j)=0, \\
    \min \begin{cases}
            \operatorname{lev}_{a,b}(i-1,j) + 1 \\
            \operatorname{lev}_{a,b}(i,j-1) + 1 \\
            \operatorname{lev}_{a,b}(i-1,j-1) + 1_{(a_i \neq b_j)}
         \end{cases} & \text{ otherwise.}
   \end{cases}$
\newline
where $1_{(a_i \neq b_j)}$ is the indicator function equal to 0 when  $a_i = b_j$ and equal to 1 otherwise, and $\operatorname{lev}_{a,b}(i,j)$ is the distance between the first $i$ characters of $a$ and the first $j$ characters of $b$.

## Levenshtein distance (continued)

- The Levenshtein distance is often referred to as the *edit distance*. 

```{r levenshteinDistanceExcercise, echo=FALSE, results='hide'}
str1 <- "AATGCTGCTT"
str2 <- "AATGCATT"
ldRes <- utils::adist(str1, str2)[1,1]
```

\exercise{What is the Levenshtein distance between \texttt{`r str1`} and \texttt{`r str2`}?}{`r ldRes`}

- Beyond bioinformatics, the Levenshtein distance is often used in computational linguistics and natural language processing. For instance, check out [*How to Write a Spelling Corrector*](http://norvig.com/spell-correct.html) by Peter Norvig.

\takehome{Levenshtein distance applies to biological sequences}

## Substitution matrix

<!-- use \scalebox in the graphicx package to resize things  --> 

- Substitution matrix describes the rate at which one character changes to other character states over time.
- Example of DNA-damage-inducible 1/2 protein (Silva *et al.*, Nature Reports, 2016): 
    * \texttt{...DTGAQTT...} (yeast)
    * \texttt{...D\textcolor{red}{S}GAQTT...} (fruit fly)
    * \texttt{...D\textcolor{red}{S}GAQ\textcolor{red}{M}T...} (human)
- The simplest substitution matrix: the Identity matrix
\scalebox{0.7}{
$\begin{bmatrix}
 1 & 0 & \cdots & 0 & 0 \\
 0 & 1 &        & 0 & 0 \\
 \vdots & & \ddots & & \vdots \\
 0 & 0 &        & 1 & 0 \\
 0 & 0 & \cdots & 0 & 1
\end{bmatrix}$
}

\question{(1) What are the advantage and disadvantage of using the identity matrix? (2) What other alternatives can you imagine?}

## Substitution matrix continued

- Log-odds matrices: we can express the probabilities of substitution with log-odds scores. The scores matrix S is defined as
    $S_{i,j} = \log \frac{p_i \cdot M_{i,j}}{p_i \cdot p_j} = \log \frac{M_{i,j}}{p_j} = \log \frac{\text{observed frequency}}{\text{expected frequency}}$
    \newline
    where $M_{i,j}$ is the probability that character $i$ transforms into character $j$, and $p_i$, $p_j$ are the frequencies of character $i$ and $j$.

- Commonly used log-odds substitution matrices
    * PAM (Point Accepted Mutation) matrix, developed by Margaret Dayhoff in the 1970s, works well to compare closely related species, \textit{e.g.} rat and mouse.
    * BLOSUM (BLOck SUbstitution Matrix), developed by Steven and Jorja G. Henikoff in early 1990s, works well for evolutionarily divergent sequences, say zebrafish and human.

## The Needleman-Wunsch algorithm

\takehome{Dynamic programming underlies the Needleman-Wunsch algorithm}

## The Smith-Waterman algorithm

\takehome{SM reaches local alignment whereas NW reaches global alignment}

# How does BLAST work

## Seuqnce query: David versus Goliath

\takehome{Sequence query is frequently used in drug discovery}

## The NCBI and command-line BLAST tool

\takehome{Anyone can query similar sequences using the BLAST tool}

## How BLAST in principle works

\takehome{BLAST is a heuristic method}

## Gumbel extreme value distribution (EVD)

\takehome{Statistical models are important components of bioinformatics}

# Probablistic modelling of biological sequences

## Model DNA/RNA sequences using Markov chains

\takehome{Markov chain is a probablistic model of biological sequences}

## Hidden Markov Models

\takehome{HMMs allow sequence discrimination}

# Conclusions

## Other sequence-analysis tasks in drug discovery

* Phylogenetic analysis of drug targets
    * Is my target present in mouse/rat/rabbit...?
    * Is the function of my target conserved in animals?
* Prediction of RNA secondary structure
* Protein sequence and structure analysis
    * Discussed in the follow-up sessions
    
## References

* Rosalind
* [Teaching RNA algorithms](http://rna.informatik.uni-freiburg.de/Teaching/) by Backofen Lab at U Freiburg, with [source code](https://github.com/BackofenLab/RNA-Playground) available on Github.
* [An Introduction to Applied Bioinformatics](http://readiab.org/)