# Biological sequence analysis {#bioseq}

```{r}
knitr::opts_chunk$set(echo=FALSE,
                      message=FALSE,
                      fig.path = "figures/02-",
                      dev = c('png','pdf'),
                      dpi = 300,
                      fig.width=7, fig.height=5)
library(DiagrammeR)
```

## What do we want to achieve today

* Learn the central dogma of molecular biology and its implications in drug discovery
* Understand how biological information is stored in DNA and protein sequences and how can they be analysed
* Understand technical terms including edit distance, Smith-Waterman algorithm, Needleman-Wunsch algorithm, Markov chains
* Know software BLAST, hmmer
* Self-learning with rosalind

## Questions

* Are there other genes other the target X with similar structures to X, that may increase the effect of the drug and/or cause side effects?
* Does it make sense to test drug candidates of target X in mouse, rat, or another model species?
* How variable are the coding sequences of the target X among the general population, and among patients of a certain disease?
* Can we design efficacious and safe RNA-based therapies that target gene X potently and specifically to reverse the disease? 


## Central dogma of molecular biology

To understand how drugs treat diseases, we need to first understand how information is stored in biological systems, how the information or its flow is distorted and interrupted in disease, and how drugs work from an information point of view.

The central dogma of molecular biology is shown in figure \@ref(fig:figCD).

```{r figCD, fig.cap="Central dogma of molecular biology.", out.width="100%", fig.height=0.8}
DiagrammeR::grViz("data/central-dogma.gv")
```

## DNA

The [1BNA](https://www.rcsb.org/3d-view/1BNA) entry of RCSB PDB database is the structure of a piece of DNA. Explore its structure with the interactive *3D view*.

<!--
```{r dna, include=FALSE, eval=FALSE}
library(Rpdb)
dna <- Rpdb::read.pdb("data/1bna.pdb.gz")
visualize(dna)
```
-->

Ech DNA nucleotide is composed of three main units:

* a pentose sguar (5-carbon monosaccharide, or deoxyribose)
* a phosphate group
* a nitrogenous base

They are connected as shown in the bottom panel of figure \@ref(DNAsegment).

```{r DNAsegment, fig.cap="Double-helix structure of DNA, nitrogenous bases pairing, and the molecular structure of a DNA molecule. From the text book [OpenStax Anatomy and Physiology](https://cnx.org/contents/FPtK1zmh@8.25:fEI3C8Ot@10/Preface), discovered through [Wikimedia](https://commons.wikimedia.org/wiki/File:0322_DNA_Nucleotides.jpg), reused under the CC license. "}
knitr::include_graphics("figures/WikimediaDNANucleotides.jpg")
```

While the phosphate group and the nitrogenous base stay constant for each nucleotide, the nitrogenous base varies. An overview of nitrogenous bases that are found in virtually all biological systems can be found in \@ref(fig:DNAbases).

<!--
nucleic acids = repeated nucleotides
nucleotide = H3PO4 + nitrogen base (also known as nitrogenous base) + pentose sugar
nucleoside = nitrogenous base + pentose suguar = nucleotide - phosphoate
-->

```{r DNAbases, fig.cap="Nitrogenous bases. From *The Basics of General, Organic, and Biological Chemistry* (*The Basics of GOB Chemistry*) by David W. Ball, John W. Hill, and Rhonda J. Scott, reused under the CC-BY-NC-SA license. Downloaded from [LibreTexts Chemistry](https://chem.libretexts.org/Bookshelves/Introductory_Chemistry/Book%3A_The_Basics_of_GOB_Chemistry_(Ball_et_al.)/19%3A_Nucleic_Acids/19.1%3A_Nucleotides) on July 31, 2019."}
knitr::include_graphics("figures/nitrogenousBases.jpg")
```

Nitrogenous bases of two anti-parallel DNA strands form pairs: Adenine (A) pairs with Thymine (T), and Guanine (G) pairs with Cytosine (C). In this way, pairs between a purine (A and G) and a pyrimidine (T and C) allow the formation of a double helix with a constant distance between the two backbone strands. The diameter of B-DNA, a common form of DNA, is about 20 Angstroms; one Angstrom is $10^{-10}$m, or 0.1 nanometer; the distance between base pairs is about 3.4 Angstrom. Apparently, pairing between two pyrimidines will take less space, and pairing between two purines will take larger space; in addition, hybrogen bond formation does not allow the pairing between two pyrimidines or purines. Readers that are interested in the physics and chemisty of DNA are encouraged to consult introductory books to biochemistry, for instance *The Basics of General, Organic, and Biological Chemistry* [@ball_basics_2012], which is shared under the Common Creative lisence.

How can we effectively represent DNA sequences? ATGCGTC

### The human genome

The human genome consists of about three billion DNA base pairs in germ cells - eggs and sperms - and twice of that in somatic cells. They are organized in 23 chromosome pairs in cell nuclei, and a small DNA molecule found within mitochondria.

Among the three billion DNA base pairs, about 0.1% are different between individuals. Compared with our closest living relatives, such as chimpanzees and bonobos, only about 4% of the genome is different.

By end of 2018, scientists around the world have sequenced the diploid genomes of over a million individuals. 

Protein-coding sequences account for only a very small fraction of the genome, approximately 1.5%. The rest is associated with non-coding RNA genes, regulatory DNA sequences, and other elements. 

Protein-coding genes are transcribed into messenger RNAs (mRNAs). The transcription is regulated at many levels, starting from the transcriptional regulation. Transcriptional regulation is achieved by interaction between DNA sequences in the gene regulatory region and proteins that are dedicated to regulate gene expression, which in turn are transcribed from other loci of the genome, translated, and exported into nuclei. 

The gene regulatory region usually contain stretches of DNA sequences that have certain functions, which are known as regulatory elements. The core promoter, a DNA stretch between approximately -40 bp and +40 base pairs of transcriptional starting site (TSS), is a regulatory element that controls basal expression. The proximal promoter elements, usually less than 1 kilobase (kb) upstream of the gene, contain short, 6-12 bases DNA sequences that are recognized by transcription factors, a class of proteins that bind specifically to such short sequences known as motifs. Mathematically, motifs are often represented as position weight matrix (PWM) and visualized in sequence logos (web logos). Transcription factors bind to DNA and collaborate with each other to regulate gene expression. In the protein section below, we will examine how transcription factor can be identified from the sequences.

Besides proximal promoter elements, other regulatory elements can exist to regulate gene expression. Enhancers and silencers regulate gene expression positively and negatively, respectively, independent of both the distance from and orientation relative to the promoter. Insulators block genes from being affected by the transcriptional activity of neighbouring genes. Bioinformaticians use the same tools that we introduced earlier for transcription factor binding motifs, position weight matrix and web logos, to describe and visualize these elements. 

Below we have a closer look at these two concepts.

#### Position weight matrix

#### Web logos

## RNA

Almost the same as DNA, but with ribose as the pentose sugar instead of deoxyribose. And it exists mostly as single strand, or forming helix with DNA.

How can we effectively represent RNA sequences? AUGUUC

## Protein

How can we effectivel represent protein sequences?

## The central dogma, revisited

## Drugs target either the nodes or the edges of the graph

## Mathematical concepts and models

### Levenshtein distance (edit distance)

### Substitution matrix

### Linear programming and the Needleman-Wunsch algorithm

### Markov markov models

### Profile hidden Markov models

(the following paragraph is almost a verbatim copy of hmmer's user guide, which is a masterpiece of writing.] Most protein sequences are composed from a relatively small number of ancestral protein domain families. The sampling of common protein domain families has become comprehensive and deep, while raw sequence data continues to accumulate explosively. It has become advantageous to compare sequences against all known domain families, instead of known sequences.

The idea is to identify subunits of protein sequences and annotate them. This resembles speech recognition, where user's voice is compared with a prebuilt dataset of statistical models of common words and phonemes. Using machine learning techniques, each statistical model is trained on large datasets of examples spoken by different speakers in different accents. Similarly, for each protein domain family, there are thousands of known homologs that can be aligned into deep multiple sequence alignments. Sequence alignments reveal a specific pattern of evolutionary conservation particular to that domains' structure and function. These patterns can be captured by probabilistic models.

These probabilistic models are called profile hidden Markov models, sometimes called profile HMMs or just profiles. They model protein and DNA sequence domain families.

Profiles can be used for

* Sensitive homology searches: build a profile based on multiple sequence alignment, and search the profile against a sequence database
* ... even for single sequence queries
* Automated annotation of protein domains
* Curated collections of deep multiple alignments

Assumptions and limitations

* Profile HMMs are primary sequence consensus models. They assume that the residue at a particular position is independent of the residues at all other positions. They ignore higher-order correlations. For instance, profile HMMs are often not good models of structural RNAs, because there base pairs are correlated.
* Lacks an explicit model of the phylogenetic relationships between a set of homologous seqeunces.

## Software

### BLAST

### hmmer

### Biopython

## Resources for further learning

* Rosalind
* [Teaching RNA algorithms](http://rna.informatik.uni-freiburg.de/Teaching/) by Backofen Lab at U Freiburg, with [source code](https://github.com/BackofenLab/RNA-Playground) available on Github.
* [An Introduction to Applied Bioinformatics](http://readiab.org/)
