# Biological sequence analysis {#bioseq}

```{r}
knitr::opts_chunk$set(echo=FALSE,
                      message=FALSE,
                      fig.path = "figures/02-",
                      dev = c('png','pdf'),
                      dpi = 300,
                      fig.width=7, fig.height=5)
library(DiagrammeR)
```

## What do we want to achieve today

* Learn the central dogma of molecular biology and its implications in drug discovery
* Understand how biological information is stored in DNA and protein sequences and how can they be analysed
* Understand technical terms including edit distance, Smith-Waterman algorithm, Needleman-Wunsch algorithm, Markov chains
* Know software BLAST, hmmer
* Self-learning with rosalind

## Questions

* Are there other genes other the target X with similar structures to X, that may increase the effect of the drug and/or cause side effects?
* Does it make sense to test drug candidates of target X in mouse, rat, or another model species?
* How variable are the coding sequences of the target X among the general population, and among patients of a certain disease?
* Can we design efficacious and safe RNA-based therapies that target gene X potently and specifically to reverse the disease? 


## Central dogma of molecular biology

To understand how drugs treat diseases, we need to first understand how information is stored in biological systems, how the information or its flow is distorted and interrupted in disease, and how drugs work from an information point of view.

The central dogma of molecular biology is shown in figure \@ref(fig:figCD).

```{r figCD, fig.cap="Central dogma of molecular biology.", out.width="100%", fig.height=0.8}
DiagrammeR::grViz("data/central-dogma.gv")
```

## DNA

The [1BNA](https://www.rcsb.org/3d-view/1BNA) entry of RCSB PDB database is the structure of a piece of DNA. Explore its structure with the interactive *3D view*.

```{r dna}
library(Rpdb)
dna <- read.pdb("data/1bna.pdb.gz")
```

Ech DNA nucleotide is composed of three main units:

* a 5-carbon monosaccharide (deoxyribose)
* a phosphate group
* a nitrogenous base

```{r DNAbases, fig.cap="Nitrogenous bases"}
knitr::include_graphics("figures/nitrogenousBases.jpg")
```


## RNA

## Protein

## Drugs target either the nodes or the edges of the graph

## Mathematical concepts and models

### Levenshtein distance (edit distance)

### Substitution matrix

### Linear programming and the Needleman-Wunsch algorithm

### Markov and hidden markov models

## Software

### BLAST

### hmmer

### Biopython

## Resources for further learning

* Rosalind
* [Teaching RNA algorithms](http://rna.informatik.uni-freiburg.de/Teaching/) by Backofen Lab at U Freiburg, with [source code](https://github.com/BackofenLab/RNA-Playground) available on Github.
* [An Introduction to Applied Bioinformatics](http://readiab.org/)
