# Biological sequence analysis {#bioseq}

## What do we want to achieve today

* Know the central dogma of molecular biology
* Understand how biological information is stored in DNA and protein sequences and how can they be analysed
* Understand technical terms including edit distance, Smith-Waterman algorithm, Needleman-Wunsch algorithm, Markov model
* Know software BLAST, hmmer
* Know self-learning algorithm rosalind

## Questions

* Are there other genes other the target X with similar structures to X, that may increase the effect of the drug and/or cause side effects?
* Does it make sense to test drug candidates of target X in mouse, rat, or another model species?
* How variable are the coding sequences of the target X among the general population, and among patients of a certain disease?
* Can we design efficacious and safe RNA-based therapies that target gene X potently and specifically to reverse the disease? 

## Background information

### Central dogma of molecular biology

Drugs target either the nodes or the edges of the graph.

### DNA, RNA, and proteins

## Mathematical concepts and models

### Levenshtein distance (edit distance)

### Substitution matrix

### Linear programming and the Needleman-Wunsch algorithm

### Markov and hidden markov models

## Software

### BLAST

### hmmer

### Biopython

## Resources for further learning

* Rosalind
* [Teaching RNA algorithms](http://rna.informatik.uni-freiburg.de/Teaching/) by Backofen Lab at U Freiburg, with [source code](https://github.com/BackofenLab/RNA-Playground) available on Github.
* [An Introduction to Applied Bioinformatics](http://readiab.org/)
