# Biological sequence analysis {#bioseq}

```{r}
knitr::opts_chunk$set(echo=FALSE,
                      message=FALSE,
                      fig.path = "figures/02-",
                      dev = c('png','pdf'),
                      dpi = 300,
                      fig.width=7, fig.height=5)
library(DiagrammeR)
```

## Goals today

* The central dogma of molecular biology and its implications in drug discovery
* Dynamic programming and (hidden) Markov chains in biological sequence analysis
* Know software BLAST, hmmer, biopython, and resources including rosalind

## Typical questions in drug discovery that call for biological sequence analysis

* Are there other genes other the target X that is similar to X, that may increase the effect of the drug and/or cause side effects when the drug interferes with their functionaltiies?
* Is it appropriate to test drug candidates of target X in mouse, rat, cyno monkey, or another model species?
* How variable are the coding sequences of the target X among the general population, and among patients of a certain disease?
* Can we design efficacious and safe RNA-based therapies that target the transcript of X potently and specifically to reverse the disease? 
* Can we design efficacious and safe gene therapy that specifically increases or decreases expression of target X?

## Central dogma of molecular biology (5 min)

To understand how drugs treat diseases, we need to first understand how information is stored in biological systems, how the information or its flow is distorted and interrupted in disease, and how drugs work from an information point of view.

The central dogma of molecular biology is shown in figure \@ref(fig:figCD).

```{r figCD, fig.cap="Central dogma of molecular biology.", out.width="100%", fig.height=0.8}
DiagrammeR::grViz("data/central-dogma.gv")
```

## DNA (5min)

The [1BNA](https://www.rcsb.org/3d-view/1BNA) entry of RCSB PDB database is the structure of a piece of DNA. Explore its structure with the interactive *3D view*.

<!--
```{r dna, include=FALSE, eval=FALSE}
library(Rpdb)
dna <- Rpdb::read.pdb("data/1bna.pdb.gz")
visualize(dna)
```
-->

Each DNA nucleotide is composed of three main units:

* a pentose sguar (5-carbon monosaccharide, or deoxyribose)
* a phosphate group
* a nitrogenous base

They are connected as shown in the bottom panel of figure \@ref(DNAsegment).

```{r DNAsegment, fig.cap="Double-helix structure of DNA, nitrogenous bases pairing, and the molecular structure of a DNA molecule. From the text book [OpenStax Anatomy and Physiology](https://cnx.org/contents/FPtK1zmh@8.25:fEI3C8Ot@10/Preface), discovered through [Wikimedia](https://commons.wikimedia.org/wiki/File:0322_DNA_Nucleotides.jpg), reused under the CC license. "}
knitr::include_graphics("figures/WikimediaDNANucleotides.jpg")
```

While the phosphate group and the nitrogenous base stay constant for each nucleotide, the nitrogenous base varies. An overview of nitrogenous bases that are found in virtually all biological systems can be found in \@ref(fig:DNAbases).

<!--
nucleic acids = repeated nucleotides
nucleotide = H3PO4 + nitrogen base (also known as nitrogenous base) + pentose sugar
nucleoside = nitrogenous base + pentose suguar = nucleotide - phosphoate
-->

```{r DNAbases, fig.cap="Nitrogenous bases. From *The Basics of General, Organic, and Biological Chemistry* (*The Basics of GOB Chemistry*) by David W. Ball, John W. Hill, and Rhonda J. Scott, reused under the CC-BY-NC-SA license. Downloaded from [LibreTexts Chemistry](https://chem.libretexts.org/Bookshelves/Introductory_Chemistry/Book%3A_The_Basics_of_GOB_Chemistry_(Ball_et_al.)/19%3A_Nucleic_Acids/19.1%3A_Nucleotides) on July 31, 2019."}
knitr::include_graphics("figures/nitrogenousBases.jpg")
```

Nitrogenous bases of two anti-parallel DNA strands form pairs: Adenine (A) pairs with Thymine (T), and Guanine (G) pairs with Cytosine (C). In this way, pairs between a purine (A and G) and a pyrimidine (T and C) allow the formation of a double helix with a constant distance between the two backbone strands. The diameter of B-DNA, a common form of DNA, is about 20 Angstroms; one Angstrom is $10^{-10}$m, or 0.1 nanometer; the distance between base pairs is about 3.4 Angstrom. Apparently, pairing between two pyrimidines will take less space, and pairing between two purines will take larger space; in addition, hybrogen bond formation does not allow the pairing between two pyrimidines or purines. Readers that are interested in the physics and chemisty of DNA are encouraged to consult introductory books to biochemistry, for instance *The Basics of General, Organic, and Biological Chemistry* [@ball_basics_2012], which is shared under the Common Creative lisence.

We let `A` denote adenine, `T` thymine, `G` guanine and `C` cytosine. Thus any DNA stretch can be represented by a character string with the four elements. For instance, the following stretch is the beginning of the mRNA encoding part of the BRCA2 gene: `ATGCCTATTGGATCC`, and the following stretch is the end of the part: `AAAAAATATATCTAA`.

The basepair matching in DNA is known as the Watson-Crick pair matching, to honour the two scientists who contributed to solving the DNA structure, with inputs from other scientists. The DNA structure was solved by X-ray crystallography. Mathematics contributes to X-ray crystallography in at least two important aspects. The study of *space groups* tells what periodic configurations are possible in the 3D space. In fact, they are often called *crystallographic groups* by mathematicians (see AMS feature column by Tony Philips). The other connection is the fact that when a monochromatic X-ray diffracts off a crystal, it performs in fact part of a mathematical operation: the Fourier transform. When the incidence angle is varied, the complete transform is produced. More about this later.

## RNA (3 min)

Almost the same as DNA, but with ribose as the pentose sugar instead of deoxyribose. And it exists mostly as single strand, or forming helix with DNA. In most organisms, uracil (U) is used in RNA instead of thymine (T).

Similarly, we can let `A`, `G`, `C`, and `U` denote adenine, guanine, cytosine and uracil, respectively. 

Following our example above, the mRNA encoded are `AUGCCUAUUGGAUCC`, and `AAAAAAUAUAUCUAAâ€˜. Notice that while DNA often forms duplexes, RNA often is single-stranded and forms complex 3D structures.

The structures can be predicted based by free-energy minimization. By assuming the nearest neighbour model, which assumes that the free energy change for forming a base pair depends only on the sequence identities of that pair and the immediately neighboring base pairs. Likewise, free energy changes associated with forming loop regions and other
structural motifs are assumed to be independent of any nucleotide outside of the loop sequence and the sequence of the bounding base pairs. Thus, the free energy change for a given structure can be computed simply by adding up all the energies associated with forming all the base pair stacks and all the other structural motifs. The parameters were determined from optical melting experiments.

Under the hood, popular software to predict RNA structure, for instance MFOLD and ViennaRNA, use dynamic programming to calculate the predicted structure. The idea is very much like the Needleman-Wunsch algorithm that we talk about below. See the primer by Sean Eddy for more details.

From a pratical point of view, RNA structures can be found by by chemical methods such as the SHAPE method. The ability to predict and determined structures of RNAs expands our ability to use RNA as drug targets, for instance using antisense oligonucleotides or RNA-binding small molecules. More about that later.  

## Protein (8 min)

Message RNA is translated into proteins in cytoplasm of human cells. The translation follows the genetic code: each three nucleotides specify one amino acid. There are twenty-one amino acids used in human and other eukaryotes. 

Exercises: (a) translate the mRNA sequences above to protein sequences (b) verify that the genetic code is degenerate, or redundant (b) what is the implication of degeneration?

## Drugs target either the nodes or the edges of the graph (15 min)

Most drugs target either the nodes or the edges of the central-dogma graph.

* Proteins are targets of most drugs. According the statistics in [@rask-andersen_trends_2011], by 2019, more than 80% drugs act on protein targets. According to the latest statistics [@oprea_unexplored_2018], among different kinds of proteins, most targeted ones include enzymes (31%), ion channels (21%), GPCR (16%). Other much explored targets include kinase (8%), transporters (4%), nuclear receptor - a class of transcription factors - (3%) and epigenetic factors (2%). Later we will briefly introduce what do these genes in principle do. And we will examine two examples, Herceptin (chemical name: trastuzumab) and cetrizine, in detail.
* Translation inhibitors, for instance antimicrobial protein synthesis inhibitors. A prominent example are tetracyclines glycylcyclines. Tetracyclines are a series of derivatives of a basic 4-ring structure known as doxycycline. Glycylcyclines are tetracycline congers with substituents that confer broad-spectrum activity, and activity against bacteria that are resistant to other antibiotics. Tetracyclines and glycylcyclines inhibit bacterial protein synthesis by binding to the 30S bacterial ribosome and preventing access of aminoacyl tRNA to the acceptor (A) site on the mRNA-ribosome complex. See more in chapter 55 of *Goodman and Gilman's Manual of Pharmacology and Therapeutics, 2e*, [Protein Synthesis Inhibitors and Miscellaneous Antibacterial Agents](https://accesspharmacy.mhmedical.com/content.aspx?bookid=1810&sectionid=124496077). In human, mTOR inhibitors, a class of immunosuppresants, drugs that supress functions of the immune system, for instance after organ transplanation or in case of autoimmune diseases, also modulates the translation process.
* RNA are targets of a class of drugs known as antisense-oligonucleotides (ASO). Chemically, they are modified, short DNA or RNA molecules that bind to an RNA target, leading to reduced expression of the protein target. The ASO-RNA recognition works by the Watson-Crick base maching mechanism that we described before. The first antisense drug that was approved in the U.S. is fomivirsen (brand name *Vitravene*), an antiviral drug that treats one kind of virus known as cytomegalovirus retinitis (CMV) in patients with deficient immune system (immunocompromised patients), such as those with AIDS. It is a synthetic 21mer oligonucleotide with phosphorothioate linkage (S instead of O), which are resistant to degradation by nucleases, with the sequence `5'-GCG TTT GCT CTT CTT CTT CTT GCG-3'`. It blocks translation of viral mRNA by binding to the complementary sequence of the mRNA transcribed from the template segment of a key CMV gene. It was discovered at National Institutes of Health (NIH), was licensed and initially developed by Isis Pharmaceuticals, which subsequently licensed it to Novartis. Novartis withdrew the marketing authorization in the EU in 2002 and in the US in 2006.The drug was withdrawn because while there was a high unmet need for drugs to treat CMV when the drug was initially discovered and developed due to the CMV arising in people with AIDS, the development of HAART dramatically reduced the number of cases of CMV.
* Transcription: some antimicrobials and chemotherapeutic agents inhibit transcription, such as Dactinomycin (known also as actinomycin D). Some strong toxins, such as \alpha-Amanitin, also interferes with transcription. In human, drugs have been developed to specifically modulate single gene and even transcript expression patterns with the help of mathematics and informatics tools. We will examine risdiplam, a small molecule for Spinal Muscle Atrophy (SMA), a type of rare disease, in details in later part of the course.
* Reverse transcription: reverse transcriptase inhibitors, such as Zidovudine (AZT), are important components of the highly active antiretroviral therapy (HAART), a way to use multiple antiretroviral drugs to control HIV infection.
* DNA: gene-therapy and cell-therapy use biotechnologies and genome-editting to introduce exogeneous DNA, or delete/modify human DNA. Examples include chimeric activated receptors in T-cells (CAR-T) as an immunotherapy for cancer. In this therapy, T cells are isolated from human blood, either from the patient or a healthy donor. The cells are stimulated, for instance by cytokine interleukin 2 (IL-2) and anti-CD3 antibodies, to proliferate and expand to large numbers. Next, they are purified and transduced with a gene encoding the engineered chimeric antigen receptor (CAR) via a retroviral vector, which is safe due to deletion of pathogenic regions, or using the CRISPR-CAS9 technology, which we will hear more about later. Last but not least, the modified CAR-T cells are infused into the patient after the patient undergos chemotherapy to remove existing T cells. CAR-T cells have two functions: it can binds to the antigen, for instance on the tumor cell, and the binding will activates T cells, which will activate the adaptive immune system.
* DNA replication: antimicrobials and chemotherapy agents, such as topoisomerase inhibitors like quinolones, inhibit DNA replication.

Exercise: (a) Derive the reverse-complement sequence of fomivirsen. (b) it is possible to search for sequence matches in large databases of nucleotides, for instance using the BLAST algorithm. An implementation is freely available at NIH: https://blast.ncbi.nlm.nih.gov/Blast.cgi. Try to search for the RNA/protein targeted by fomivirsen.

## The human genome (10 min)

Modern drug discovery depends on understanding architecture and function of human genome, as well as genomes of other species. Some of them are very different from human genomes, for instance those of virus and bacteria. Others are in many aspects rather similar, especially major model species, including mouse, rat, rabbit, dog, pig, and non-human primates. Phylogenetics is the study of the evolutionary history and relationships among individuals or groups of roganisms (species, populations). From a genomic point of view, the phylogenetic relationship between major species is shown in the figure below.

The human genome consists of about three billion DNA base pairs in germ cells - eggs and sperms - and twice of that in somatic cells. They are organized in 23 chromosome pairs in cell nuclei, and a small DNA molecule found within mitochondria.

Among the three billion DNA base pairs, about 0.1% are different between individuals. Compared with our closest living relatives, such as chimpanzees and bonobos, only about 4% of the genome is different.

By end of 2018, scientists around the world have sequenced the diploid genomes of over a million individuals. 

Protein-coding sequences account for only a very small fraction of the genome, approximately 1.5%. The rest is associated with non-coding RNA genes, regulatory DNA sequences, and other elements. 

Protein-coding genes are transcribed into messenger RNAs (mRNAs). The transcription is regulated at many levels, starting from the transcriptional regulation. Transcriptional regulation is achieved by interaction between DNA sequences in the gene regulatory region and proteins that are dedicated to regulate gene expression, which in turn are transcribed from other loci of the genome, translated, and exported into nuclei. 

The gene regulatory region usually contain stretches of DNA sequences that have certain functions, which are known as regulatory elements. The core promoter, a DNA stretch between approximately -40 bp and +40 base pairs of transcriptional starting site (TSS), is a regulatory element that controls basal expression. The proximal promoter elements, usually less than 1 kilobase (kb) upstream of the gene, contain short, 6-12 bases DNA sequences that are recognized by transcription factors, a class of proteins that bind specifically to such short sequences known as motifs. Mathematically, motifs are often represented as position weight matrix (PWM) and visualized in sequence logos (web logos). Transcription factors bind to DNA and collaborate with each other to regulate gene expression. In the protein section below, we will examine how transcription factor can be identified from the sequences.

Besides proximal promoter elements, other regulatory elements can exist to regulate gene expression. Enhancers and silencers regulate gene expression positively and negatively, respectively, independent of both the distance from and orientation relative to the promoter. Insulators block genes from being affected by the transcriptional activity of neighbouring genes. Bioinformaticians use the same tools that we introduced earlier for transcription factor binding motifs, position weight matrix and web logos, to describe and visualize these elements. 

Given the huge size of the humen genome, mathematics and informatics tools are fundamental to understand its architecture and function. It is part of the Level 1 modelling, namely molecular modelling. In the coming two sessions, we will discuss other aspects of molecular modelling, especially drug-target modelling.

## Levenshtein distance (edit distance, 10 min)

A case study of dynamic programming

## Substitution matrix (5 min)

Replacing the edit distance by biology-driven costs

## The Needleman-Wunsch algorithm (10 min)

An extension of the Levenshtein distance

## Position weight matrix (5 min)

When many similar sequences are compared

## Web logos (5 min)

Information theory meets biological sequence

### Markov markov models (10 min)

### Profile hidden Markov models

(the following paragraph is almost a verbatim copy of hmmer's user guide, which is a masterpiece of writing.) Most protein sequences are composed from a relatively small number of ancestral protein domain families. The sampling of common protein domain families has become comprehensive and deep, while raw sequence data continues to accumulate explosively. It has become advantageous to compare sequences against all known domain families, instead of known sequences.

The idea is to identify subunits of protein sequences and annotate them. This resembles speech recognition, where user's voice is compared with a prebuilt dataset of statistical models of common words and phonemes. Using machine learning techniques, each statistical model is trained on large datasets of examples spoken by different speakers in different accents. Similarly, for each protein domain family, there are thousands of known homologs that can be aligned into deep multiple sequence alignments. Sequence alignments reveal a specific pattern of evolutionary conservation particular to that domains' structure and function. These patterns can be captured by probabilistic models.

These probabilistic models are called profile hidden Markov models, sometimes called profile HMMs or just profiles. They model protein and DNA sequence domain families.

Profiles can be used for

* Sensitive homology searches: build a profile based on multiple sequence alignment, and search the profile against a sequence database
* ... even for single sequence queries
* Automated annotation of protein domains
* Curated collections of deep multiple alignments

Assumptions and limitations

* Profile HMMs are primary sequence consensus models. They assume that the residue at a particular position is independent of the residues at all other positions. They ignore higher-order correlations. For instance, profile HMMs are often not good models of structural RNAs, because there base pairs are correlated.
* Lacks an explicit model of the phylogenetic relationships between a set of homologous seqeunces.

## Software

### BLAST

### hmmer

To construct profile HMM models, hmmer uses in essence an Expectation-Maximization (EM) algorithm. To avoid the computational bottleneck where all paths need to be considered, a dynamic programming technique, known as the forward-backward procedure, is used. Details can be found in the excellent tutorial article on HMMs by Rabiner (1989). A variant of forward-backward procedure is the Viterbi algorithm. Instead of calculating the likelihood score of a sequence, which implicitly involves all possible paths for that sequence through the model, the Viterbi algorithm computers the score of the single most likely path for the sequence.

### Biopython

## Resources for further learning

* Rosalind
* [Teaching RNA algorithms](http://rna.informatik.uni-freiburg.de/Teaching/) by Backofen Lab at U Freiburg, with [source code](https://github.com/BackofenLab/RNA-Playground) available on Github.
* [An Introduction to Applied Bioinformatics](http://readiab.org/)

## Summary

## Topics that we leave out

* Phylogenetic analysis
* Comparative genomics


