---
bibliography: 00-preface.bibtex
geometry: margin=2cm
linkcolor: "blue"
mainfont: "DejaVu Serif"
monofont: "DejaVu Sans Mono"
output:
  pdf_document:
    latex_engine: xelatex
---

# Preface {-}

This document is also the proposal for *Multiscale Modelling in Drug Discovery:
An Introduction* (working title), a book that we, Philipp Mekler and Jitao David Zhang, are writing.

## Modelling in drug discovery

Drug discovery is the activity of identifying chemical or biological entities
that potentially slow down, halt, or even reverse disease progression. Modern
drug discovery has a short history compared with that of human and human
diseases. It has, however, fundamentally changed the life of each of us.

Drug discovery has also changed itself during its development. In its early
years, drug discovery was primarily driven by trials and errors as well as
serendipity in chemistry, biology, and medicine. Today, drug discovery
represents one of the most interdisciplinary and collaborative human research
enterprises. The trial-and-error principle and serendipity still play a role,
because our knowledge in biology is much limited. Nevertheless, as the
techniques of drug discovery develop and refine, mathematical, statistical, and
computational models have become essential besides experimental methods to
predict drug targets, to identify drug molecules, and to describe how drug
works.

What are these models? A mathematical model describes the state and dynamics of
a system. A statistical model represents assumptions about sample data, which
allows us to use data analysis to infer the distribution of variables of
interest. A computational model implements logic in procedures, which simulate
the behaviour of a system.

Examples of these models are introduced throughout the book. There are different
views in the literature on the philosophy of science about whether statistical
and computational models are special forms of mathematical models or they are
distinct forms of conceptual modelling. For our purpose though, it suffices that
we refer to all of them as models and specify the type only in case of
ambiguity.

We apply models for two reasons mainly. First, they allow us to study the human
body as a biochemical system. Since the system is large and complex, we usually
consider part of it. Examples of such sub-systems include the brain suffering
from the Alzheimer's Disease, the lung infected by the coronavirus, and
malignant melanoma, a skin cancer that originates from pigment-producing cells
known as melanocytes. From this perspective, health and diseases are the states
of the system. Models are used to characterize the different states: what
variables are correlated with the states? Which variables are causal,
manipulating of which may worsen or cure the disease? For simplicity, we say
that models serve *disease understanding* in such contexts.

The other main application of models is to study body-drug interactions. While
it is feasible and sometimes helpful to ignore the human body system and model
drug molecules alone, the effectiveness of such models will finally be tested in
the body system. Therefore, even if we start by modelling drug molecules, we
will end up modelling body-drug interactions because they form the mechanistic
basis of the therapeutic use of drugs.

From a system's perspective, the goal of drug discovery is to identify a
perturbation or a combination of perturbations that switches the state of the
sub-system from 'disease' to 'health', while causing as little harm as possible
to the whole system. A drug's capacity of switching the state of a sub-system
from 'disease' to 'health' is defined as its *efficacy*. Its capacity of causing
harm, known as *adverse effects*, to the whole system is defined as its
*toxicity*. Alternately, its capacity of not causing adverse effects is its
*safety*. Therefore, models characterizing body-drug interactions reveal
*efficacy and safety profiles* of drug candidates. Since a successful
drug-discovery project entails favourable efficacy and safety profiles, models
serve *drug discovery* in such contexts.

It is obvious that the two types of applications depend on each other: a better
understanding of the human body system and more knowledge of diseases catalyse
new drugs with better efficacy and safety profiles. The study of body-drug
interactions offers more than chemical or biological tools that treat diseases.
The tools can be used to interrogate biological model systems of various kinds,
including cultured cells, known as *in vitro* systems, and animal models, known
as *in vivo* systems. Such studies may reveal unexpected aspects of diseases or
even new insights into fundamental biology.

In summary, modern drug discovery can be thought of as an iterative process
consisting of experiments and modelling. The starting point is a disease that we
wish to cure and some biological or chemical entities that we deem as
appropriate prototypes of drug candidates. We collect data on the disease and on
the candidate molecules from experiments. Models characterize diseases as states
of the system and encode our understanding of body-drug interactions. Informed
by these models, we use knowledge in chemistry and biology to change the
molecule in order to improve its efficacy and safety profile. In the next cycle,
more experiments further refine the drug candidate, and improved models update
our belief about how the molecule interacts with the body. The process is
repeated until a drug with desired efficacy and safety profiles is found.

This book introduces basic concepts, principles and selected tools of modelling
both for disease understanding and for efficacy and safety profiling of drug
candidates. The aim is to integrate diverse modelling approaches at different
levels into a unified, *multiscale-modelling* view of drug discovery.

## The target audience {-}

This book aims at providing undergraduate and graduate students of mathematics,
computer sciences, natural sciences, and medicine with an application-oriented,
interdisciplinary introduction to quantitative and modelling aspects of drug
discovery. It can also be used by researchers working in the fields of disease
understanding and drug discovery as a practical guide to modelling.

The book attempts to offer a broad overview of basic concepts and principles of
quantitative modelling in drug discovery. It delivers neither comprehensive
discussions nor technicalities of individual modelling methods. References to
resources dedicated to specific topics are provided at the end of each chapter.

## Why we need such a book?

Many great books cover modelling and system methods in biology, for instance
@segel_primer_2013, @haefner_modeling_2005, @ingalls_mathematical_nodate, and @murray_mathematical_2002;@voit_first_2013,
from which we draw inspirations and borrow concepts and examples. However, few
of such books focus on modelling approaches in drug discovery.

Other books give practical guide to drug discovery, for instance
@wermuth_practice_2015, @kenakin_pharmacology_2019, and
@gabrielsson_pharmacokinetic_2016, from which we benefit as well. However, they
focus on in diverse, specific aspects of drug discovery. For instance, the three
outstanding textbook dive deep in medicinal chemistry, pharmacology, and
pharmacokinetic and pharmacodynamic modelling, respectively. In this book, we
wish to build upon these efforts to present, compare, and link different modelling
practice in the drug-discovery process, with the wish to provide an overview and
to integrate them.

## Organization of the book {-}

The content of the book is organized in three parts:

1. Introduction to modern drug discovery
2. Modelling at individual levels
    1. Molecular-level modelling
    2. Cellular-level modelling
    3. Organ- and system-level modelling
    4. Population modelling and reverse translation
3. Multiscale modelling: current practice and prospects

### Part I: Introduction to Modern Drug Discovery {-}

#### Chapter One: A Brief History of Modern Drug Discovery {-}

We learn the principles of modern drug discovery by following its history
[@drews_drug_2000;@hughes_principles_2011;@gerald_drug_2013], focusing especially on how models change our way of
finding drugs. We next examine three important concepts of drug discovery that
are relevant for later chapters:

1. the choice between target-based and phenotypic drug discovery,
2. the balance between efficacy and safety, and
3. drug modalities, such as small molecules, oligonucleotides, antibodies, recombinant proteins, and vaccines.

#### Chapter Two: The Linear View of Drug Discovery {-}

The linear view of drug discovery is prevalent [@paul_how_2010]. It is found everywhere from
scientific publications to websites of pharma companies. In this chapter, we
will examine it from three perspectives: how can we interpret it? What is the
time and cost associated with each stage in the linear view? What are the
advantages and disadvantages of such a view?

Once we are familiar with the linear view, we are ready to take an alternative,
*multiscale-modelling* perspective in the next chapter. It may help us better
understand some aspects of drug discovery.

#### Chapter Three: The Multiscale-modelling View of Drug Discovery {-}

We examine models constructed at multiple levels, including molecular-level
models, cellular-level models, system- and organ-level models, and population
models [@zhang_multiscale_2020]. They are interconnected, informing and providing feedback to each other.

We will scrutinize and challenge the multiscale-modelling view: what are its
advantages and disadvantages compared with the linear view? What new insights
can it bring us?

### Part II: Modelling at Individual Levels {-}

With multiscale modelling as a mental model of drug discovery, we start our
journey of learning basic concepts, principles, and tools of modelling at
individual levels.

#### Part II.I: Molecular-level Modelling {-}

Chapter four to six focus on molecular modelling of drug targets, including DNA,
RNA, and proteins. Chapter seven to nine go on introducing molecular
modelling of drug candidates and their interactions with targets.

#### Chapter Four: The Central Dogma and Drug Targets {-}

The central dogma of molecular biology dictates the rule of information flow in
biological systems. It builds the molecular foundation of drug discovery. Drugs,
in order to treat diseases, must modulate components specified in the central
dogma [@santos_comprehensive_2017].

We learn how biological information is stored in and transferred between DNA
(the building blocks of the genome), RNA (the building blocks of the
transcriptome), and protein (the building blocks of the proteome). We examine
approved drugs by their targets in the context of the central dogma. Finally, we
introduce extensions of the classical central dogma that are relevant for drug
discovery, including post-translational modification, epigenetic modification,
and prions.

#### Chapter Five: Biological Sequence Analysis {-}

DNA, RNA, and protein sequence analysis plays an important role in drug target
identification and assessment. We introduce both deterministic and probabilistic
approaches to biological sequences analysis [@durbin_biological_1998;
@compeau_bioinformatics_2015; @caporaso_introduction_nodate], using the
Levenshtein distance, dynamic programming, and (hidden) Markov chains as
examples.

We also discuss other applications of biological sequence analysis in drug
discovery, including phylogenetic analysis and off-target prediction of
oligonucleotides.

#### Chapter Six: Drug targets and drug-target interactions {-}

In target-based drug discovery, structural information of the target is
critical. Its resolution and availability often determine how a discovery
project is designed and executed. Even for phenotypic discovery projects,
structural information can be helpful because it may allow the prediction of
potential targets and off-targets of the drug candidate
[@schlick_molecular_2010].

For DNA and RNA targets, we focus on the thermodynamic modelling of DNA/RNA and
RNA/RNA interactions. It complements sequence analysis introduced in the last
chapter for the prediction of efficacy and safety profiles of nucleotide-based
therapeutics.

For protein targets, we introduce computational and biophysical methods commonly
used to determine 3D structures. They include homology modelling, X-ray
crystallography, Nuclear Magnetic Resonance (NMR), and cryo-electron microscopy
(cryo-EM). We highlight the key models underlying these methods. Finally, we
introduce databases and software tools that store, visualize, and use structural
information.

We also introduce *in vitro* pharmacological models of quantifying drug-target
interactions, using Michaelis-Menten and the Hill function as
examples [@wermuth_practice_2015].

#### Chapter Seven: Structure-based Modelling {-}

Structure-based modelling describes and predicts binding modes between a drug
candidate and its target [@sliwoski_computational_2014]. It is essential for
target-based drug discovery because the binding profile largely determines the
efficacy and safety profiles of a molecule. Structure-based modelling and
medicinal chemistry experiments complement and inform each other to improve the
affinity and specificity of a molecule iteratively. We examine the concepts and
techniques of structure-based modelling for computer-aided drug design. We also
discuss further applications including toxicity prediction and virtual
screening.

#### Chapter Eight: Ligand-based Modelling {-}

Complementary to structure-based modelling, ligand-based modelling uses
cheminformatics tools to describe chemical molecules and to infer their
physicochemical and biological properties. We highlight the key concepts and
approaches to ligand-based modelling, its commonality with and differences from
structure-based modelling, and its applications in drug discovery [@brown_silico_2015].

#### Chapter Nine: Antibody Design {-}

Antibodies are proteins that bind to and modulate the function of other
proteins. Engineered human antibodies represent an important class of drugs [@jain_biophysical_2017]. We
introduce key models that are critical for therapeutic antibody design. We also
review tools and databases that assist antibody design.

### Part II.II: Cellular Modelling {-}

#### Chapter Ten: Network Analysis {-}

Graph, or network, is a collection of nodes (also known as vertices) and edges
that connect them. It is an important mathematical model that links
molecular-level modelling with cellular-level modelling. We examine selected
deterministic and probabilistic approaches to chemical and biological networks
analysis. Examples include statistical methods, graph-theory-based methods,
rule-based methods, and methods based on either ordinary differential equations
(ODEs) or stochastic differential equations (SDEs).

#### Chapter Eleven: DNA- and RNA-Sequencing {-}

Genomics and transcriptomics studies with high-throughput DNA and RNA sequencing
greatly expanded our ability to understand diseases. They are also used to
assess the efficacy and safety profiles of drug candidates. With the recent
progress, it is possible to profile DNA or RNA molecules in a single cell, in
sub-compartments such as the nucleus, or in cells that are organized in
two-dimensional or three-dimensional structures.

We introduce the principles of selected sequencing technologies, including the
Illumina next-generation sequencing and Nanopore sequencing. We discuss key
concepts and models to analyse the sequencing data. And we examine their
applications in drug discovery. In particular, we will introduce selected
approaches to molecular phenotyping [@drawnel_molecular_2017], single-cell
sequencing data analysis [@madler_besca_2020], and multimodal analysis, including machine learning
[@badillo_introduction_2020]. Results of these analyses hold promise for
knowledge about diseases and body-drug interactions on the level of individual
cells and between-cell interactions.

#### Chapter Twelve: Proteomics and Metabolomics {-}

Proteomics and metabolomics describe abundance, modulation, and homeostasis of
proteins and metabolites, respectively. Study of the healthy baseline and
pathologically disturbed states of these systems reveal a wealth of drug
target and mechanistic information of how drug works. Proteomics and
metabolomics technologies find a wide range of applications in drug discovery,
from quantification of drug-target binding to accurate quantification of
biomarkers.

We introduce the principles of two widely used technologies, mass spectrometry
and flow cytometry. We examine the practice of data analysis and modelling.
Finally, we discuss the applications of proteomics and metabolomics in drug
discovery.

#### Chapter Thirteen: Biological Imaging {-}

Biological imaging complements the omics technologies introduced in the last two
chapters for studies of body-drug interactions. Application of imaging spans a
wide range between molecular modelling and population modelling.

Here, we first give a brief overview of imaging technologies used in drug
discovery [@scheeder_machine_2018;@rudin_molecular_2003]. We next focus on cellular-level imaging, in particular high-content
microscopy [@feng_multi-parameter_2009]. We introduce machine-learning based methods of
biological imaging data analysis [@meijering_biological_2006;@mullard_machine_2019].

### Part II.III: Organ- and System-level Modelling {-}

#### Chapter Fourteen: Pharmacokinetic and Pharmacodynamic modelling {-}

Pharmacokinetic (PK) modelling describes and predicts what the body does to the
drug. Pharmacodynamic (PD) modelling, on the other hand, describes and predicts
what the drug does to the body [@mortensen_introduction_2008;@gabrielsson_pharmacokinetic_2016].

We introduce the basic models of PK and PD modelling, including a
one-compartment PK model, a two-compartment PK model, and a joint PK-PD model.
We examine how mathematical modelling and statistical data analysis help to
characterize PK and PD profiles of drug candidates.

#### Chapter Fifteen: Physiologically Based Pharmacokinetic Modelling {-}

Physiologically based pharmacokinetic (PBPK) modelling represents our knowledge
of human physiology in mathematical structures. It can be seen as a natural
extension of the PK and PD models introduced in the last chapter. PBPK models
are applied for a wide range of purposes from animal-human translation to
paediatric dose selection [@jones_basic_2013; @kuepfer_applied_2016].

We introduce the basic concepts and principles of PBPK modelling and tools to
construct and analyse the models.

#### Chapter Sixteen: Quantitative Systems Pharmacology Models {-}

Quantitative systems pharmacology (QSP) models, or systems models for short,
integrates preclinical mechanistic data with PK and PD profiles and
patient-specific characteristics collected in clinical studies. They can
identify mechanisms of actions of drugs, which may offer insight for compound
optimization or even lead to the identification of new targets. They can also
assess safety liabilities under varying dosing regimens in different populations
quantitatively [@knight-schrijver_promises_2016; @leil_quantitative_2014;
@helmlinger_drug-disease_2017].

We introduce principles and basic applications of system models to investigate
both efficacy and safety profiles of drug candidates. We will learn from systems
models developed in drug discovery and development projects.

### Part II.IV: Population Modelling And Reverse Translation {-}

#### Chapter Seventeen: Randomized Controlled Trials {-}

Randomized controlled trial (RCT) is the gold standard of measuring the
effectiveness of an intervention or a treatment. It is used in the form of
clinical trials to examine the efficacy and safety profiles of a drug in a
defined patient population [@evans_clinical_2010].

We learn key concepts and principles of designing a randomized controlled trial,
including statistical power calculation, randomization, blinding, and causal
inference. We also examine the common forms of clinical trials in drug discovery.

#### Chapter Eighteen: Population Modelling {-}

Population modelling informs common trends and individual variabilities of PK
and PD profiles of a drug. It is used to infer associations between individual
variables (age, sex, genotype, *etc.*) [@mould_basic_2012; @mould_basic_2013;
@upton_basic_2014] and PK/PD profiles in clinical
pharmacology [@sheiner_population_1999]. We introduce the basic concepts of population modelling and its
techniques, in particular the statistical model known as non-linear mixed-effect
modelling.

#### Chapter Nineteen: Real-World Data and Reverse Translation {-}

Real-World Data (RWD) refers to data collected about patients, including the
personal data, omics data, medical records, and other data types that can be
potentially used to infer efficacy and safety profiles of drugs. Classically,
real-world data, like clinical-trial data, are analysed 'for-purpose' to serve
the development of the drug product for which the data were collected.
Recently, however, more researchers realize the potential of such data to inform
both preclinical discovery and clinical research [@gibbs_bedside_2018;
@miksad_harnessing_2018; @schneeweiss_real-world_2018].

The term *reverse translation* was coined to indicate the practice of informing
drug discovery with insights from drug development and real-world evidence. It
complements *forward translation*, which indicates the common drug discovery
and development workflow starting from animal or *in vitro* disease models to
patient populations. We examine key modelling approaches used in real-world
data analysis and reverse translation as well as both opportunities and
challenges.

### Part III: Multiscale Modelling: Current Practice and Prospects {-}

#### Chapter Twenty: Examples of Multiscale Modelling in Drug Discovery {-}

We examine state-of-the-art practice and emerging patterns of multiscale
modelling in drug discovery [@walpole_multiscale_2013; @amaro_multiscale_2018]. The examples come from both industrial and academic
research.

In particular, the study of (i) excitable systems, (ii) stem-cell drive complex
biosystems, (iii) nanoparticle delivery, (iv) immune systems, and (iv)
host-pathogen interactions lend themselves to the tools of multiscale
modelling. The following examples are presented in some detail:

* Discovery of Vemurafenib
* Discovery of Sofosbuvir
* Discovery of Risdiplam
* Discovery of halicin as a new antibiotic

#### Chapter Twenty-one: Outlook of Multiscale Modelling in Drug Discovery {-}

In the final chapter, we discuss the outlook of multiscale modelling in drug discovery.

We first discuss future developments at individual levels.

* At the molecular level, molecular simulation of longer time frames may reveal
  rare and unexpected structures that are important for drug design.
* At the cellular level, integration of network modelling, multimodal omics,
  and imaging may associate different aspects of cell biology and integrate
  them into a unified view.
* At the organ and system level, better models to simulate human physiology,
  such as induced pluripotent stem cells and micro-physiological systems, will
  allow better predictions of PK and PD profiles. Patient data and knowledge of
  genotype-phenotype association collected from population studies may allow
  personalized PK and PD models and individualized dosing.
* At the population level, digital biomarkers and integration of real-world
  data may allow non-invasive and continuous monitoring of response to drugs.
  Such data may change our look at traditional 'normal physiological values',
  requiring the definition of both individual- and population-based normals.

We list key open questions about integrating models at different levels. We
study the success of multiscale modelling in two other scientific fields,
molecular modelling in chemistry and material science in physics. Our hope is to
speculate how drug discovery may learn and benefit from their approaches.

Finally, we consider how interdisciplinary multiscale modelling may shape drug
discovery in the future. We focus on two aspects: computer-enhanced human
cognition, and uncertainty- and risk-aware decision-making. Mathematical,
statistical, and computational models, when unified and integrated, may help us
identify, build, and test better drug-like molecules faster and cheaper. They
may also help us make better decisions that are optimized given uncertainty and
risk. If implemented, these measures may revolutionize the way we find drugs in
the future.

## Acknowledgement {-}

We would like to thank many mentors, colleagues, and friends who have helped
and supported us.

## Licensing {-}

<a rel="licence" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="figures/CC-BY-SA-4.0.png" /></a><br />The online version of this book is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative Commons Attribution-ShareAlike 4.0 International Licence</a>.

## References

