# Genomics, transcriptomics, and proteomics {#omics}

## What we want to achieve today

* Understand the basics of omics technologies, especially in the light of central dogma and biological sequence analysis
* Know technical terms including De Bruijn graphs, Burrows-Wheeler transformation
* Have a first understanding of graph models with the example of RSEM
* Have a first understanding of linear models and generalised linear models with the example of edgeR

We describe applications of omics in disease understanding and drug discovery, especially in the light of underlying mathematical concepts.

## Questions

## Background information

### Principles of DNA and RNA sequencing

Sanger sequencing represents the first generation of DNA sequencing technique.

Second generation sequencing, sometimes called next-generation sequencing, comes in several variants. One variant is Illumina sequencing, sometimes known as sequencing by synthesis. DNA sequences to be sequenced, together with ingridients that are required for the synthesis, such as enyzmes, are dispersed into droplets. These seqeunces can be potentially modified, for instance, with the addition of a short strech of artificial nucleotides as *barcodes*. In each round of synthesis, a nucleotide is added to the newly synthesised strand. The addition of the nucleotide is detected by sensors that recognize nucleotides based on their illuminant modifications. A camera takes pictures of thousands of droplets and image analysis is used to read the nucleotide added at each step. At the end of the sequencing process, the nucleotide sequence of the original template is constructed by concatenateing the detected nucleotides together. They are saved in FASTQ files.

A variant is not to sequence the whole stretch of the template, but rather only the two ends of it. This will give pair-end (PE) reads.

RNA can be reverse trascribed into DNA. Therefore RNAs can be sequenced just as DNAs, with the difference that uracil (*U*) replaces thymine (*T*).

### Genomics

The genetic information, for instance DNA sequence in our genomes, varies between species, between individuals, and between different cells in our body. 

Chimpanzees and human share 96% identical DNA sequences in their genomes. Mice and human share 85% and 50% identical DNA sequences in protein-coding and non-coding regions. 

Between human individuals, 99.9% DNA sequences are identical. This means that out of six billion DNA bases in somatic cells, about a few million bases are different.

How about DNA variations between cells in a human body?

### Transcriptomics

### Proteomics

### Next-generation sequencing

### Mass spectrometry based protein quantification

## Mathematical concepts and models

### De Bruijn graph

### Graph model

### Linear model

### Negative binomial distribution

### Dimension reduction

#### Principal component analysis

#### Exploratory factor analysis

Given gene expression and other high-dimensional data, we want to understand
*why* they are so. For instance, what transcription factors and other mechanisms
of gene expression are work to shape the gene expression data. While the
underlying ground truth may be too complex to be known, we may achieve an
approximate understanding if we allow some assumptions. Factor analysis is such
a method that allows us to achieve such understandings while making minimal
assumptions.

Factor analysis describes the variability among observed, correlated variables
by fewer unobserved variables called factors. In the context of gene expression
profiling, individual genes can be thought of as correlated variables and their
expression in a number of samples are observed values. Factor analysis
identifies unobserved factors, which are generally fewer compared to the number
of genes, that explain the variations of genes.

For instance, let us assume that we are interested in 10 genes and we measure
their expression in the blood of 100 individuals. We have the hypothesis that
these genes are under control of two transcription factors, `TF1` and `TF2`,
neither of which is directly observed. If each individual is chosen randomly
from a large population, each each individual's expression of 10 genes are
random variables. We hypothesize that in a particular individual, the expression
of any gene is a linear sum of the factors. 

$$
g_i ~ N(m, sigma^2)
m = \lambda_i1 TF_1j + \lambda_i2 TF_2j
i in 1,2,...,10
j in 1,2,...,100
$$

$TF_1j$ and $TF_2j$ are activities of the two transcription factors that we wish
to infer for each individual. $lambda_i1$ and $lambda_i2$ are the *factor
loadings* associated with gene $i$. Different genes have different factor
loadings, which reflect the aptitude of the two transcription factors for
regulating them.

Two individuals assumed to have the same activities of $TF_1$ and $TF_2$ may
have different expression in any gene, because (1) the equation above only makes
predictions based on the population, while individuals may have aberrant
expression profiles due to many factors (disease status, genetic background,
etc.), and (2) the measurement may contain errors. Both types of differences are
collectively called *errors*, which means the amount by which an individual, as
measured, differs from what is predicted. Because many factors may contribute to
the errors, their sum can be well modelled by the Gaussian distribution, thanks
to the Central Limit Theorem.

The observed data that go into factor analysis is expression of 10 genes of each
of the 100 individuals, a matrix of 1000 numbers. The output are the factor
loadings (how two transcription factors regulating the genes) and activity
levels of the transcription factors of each individual.

## Resources for further learning

### Genomics

### Transcriptomics

### Mass spectrometry

[Khan Academy](https://www.khanacademy.org/science/chemistry/atomic-structure-and-properties/mass-spectrometry/a/isotopes-and-mass-spectrometry)

Meija, Juris. “Mathematical Tools in Analytical Mass Spectrometry.” Analytical and Bioanalytical Chemistry 385, no. 3 (June 1, 2006): 486–99. https://doi.org/10.1007/s00216-006-0298-4.

[Factor Analysis](https://en.wikipedia.org/wiki/Factor_analysis), Wikipedia
